### PR TITLE
environment.yml: install pytorch with gpu support

### LINKED ISSUE #529: Conda installs a cpu version of pytorch and torchvision
Conda installation (tested on Ubuntu 20.04) installs a cpu version of pytorch and torchvision. In my own project, the only remedy I've found was to [hardcode the pytorch package](https://github.com/remtav/geo-deep-learning/blob/222-stac-item-input/environment.yml#L9) to use. Poor solution. Hoping to find an alternative.

Related to #527 

result from `conda list` in torchgeo environment:
ksnip_20220503-162044
ksnip_20220503-162055

### PR DESCRIPTION
**Pull request by remtav**
fixes #529 
see #540

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ channels:
   - esri  # for pycocotools. Remove when https://github.com/conda-forge/pycocotools-feedstock/issues/16 is solved
   - open3d-admin  # for open3d
 dependencies:
-  - cudatoolkit
+  - cudatoolkit<=11.5
```
[the remark this answers is not in the document]
Comment (adamjstewart) [reply environment.yml:8]: > Not sure how this affects the CI though.

This file isn't used by CI. If this solution works I like it better than pinning.

[the remark this answers is not in the document]
Comment (weiji14) [reply environment.yml:8]: Yeah, as you mentioned in https://github.com/microsoft/torchgeo/issues/529#issuecomment-1136484142, there's no point in specifiying cudatoolkit explicitly in this environment.yml. For GitHub Actions CI which doesn't have a GPU, cudatoolkit definitely won't be used, but need to try to be sure.

[the remark this answers is not in the document]
Comment (adamjstewart) [reply environment.yml:8]: For GitHub Actions, the only thing Conda is used for is to install packages without Windows wheels (fiona, rasterio), we don't use Conda to install PyTorch.

[the remark this answers is not in the document]
Comment (weiji14) [reply environment.yml:8]: Right, then I'd recommend removing `cudatoolkit` entirely then, no point in having it if we want to allow CPU-only installs. Could you push the suggested change above?

I'm looking into the Windows build failures you mentioned at https://github.com/microsoft/torchgeo/pull/543#issuecomment-1139811728, see diff at https://www.diffchecker.com/7lbnhaAb (between previous successful CI run and failed run). Nothing obvious sticking out currently, the crash is very weird and I don't have a Windows computer to debug.

**Review by adamjstewart — APPROVED** [review]
LGTM. Can someone confirm that this now works as intended?

**Comment by weiji14** [comment]
Should have changed the PR title before merging :sweat_smile:

**Comment by adamjstewart** [comment]
Oh well, commit messages aren't super important to me, issue titles are more important because that's what I look for when writing release notes. I think this change is minor enough that it won't be in the release notes anyway.

**Pull request merged**